# Kalman node parameters for ROS 2

kalman_parameters: &kalman_parameters
  ros__parameters:
    imu_version: 1
    imu_offset: 0.0
    dvl_max_velocity: 0.5

    state_vector: [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]

    cov_matrix: [
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.
    ]

    Q: [
      0.0001, 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0.01, 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0.01, 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0.1, 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0.0001, 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0.1, 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0.00015, 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0.00009, 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0.1, 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0.001, 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.01, 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.01
    ]

    A_imu: [
      1., 0., 0., 0., 0., 0., 0.005, 0., 0., 0., 0., 0.,
      0., 1., 0., 0., 0., 0., 0., 0.005, 0., 0., 0., 0.,
      0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 1., 0., 0., 0., 0., 0., 0.005, 0., 0.,
      0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0.005, 0.,
      0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1.
    ]

    dt_dvl: 0.2
    R_dvl: [0.0001, 0., 0., 0., 0.0001, 0., 0., 0., 0.001]
    H_dvl: [
      0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 1., 0., 0., 0.
    ]

    dt_imu: 0.005
    R_imu: [0.01, 0., 0., 0., 0.01, 0., 0., 0., 0.01]
    H_imu: [
      0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 1., 0., 0., 0., 0., 0., 0.
    ]

    dt_depth: 0.25
    R_depth: [0.01, 0., 0., 0., 0.01, 0., 0., 0., 0.01]
    H_depth: [
      0., 0., 1., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
      0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.
    ]



    offset:
      x: 0.
      y: 0.
      z: 45.

"/bruce/slam/kalman":
  <<: *kalman_parameters

kalman:
  <<: *kalman_parameters


